src/domain/entities/figma-design-identity.test.ts

```diff
@@ -0,0 +1,139 @@
+import { FigmaDesignIdentity } from './figma-design-identity';
+import {
+	MOCK_FIGMA_FILE_IDENTITY,
+	MOCK_FIGMA_FILE_KEY,
+	MOCK_FIGMA_NODE_ID,
+	MOCK_FIGMA_NODE_IDENTITY,
+} from './testing';
+
+describe('FigmaDesignIdentity', () => {
+	describe('constructor', () => {
+		it('should throw when fileKey is empty string', () => {
+			expect(() => new FigmaDesignIdentity('')).toThrow();
+		});
+
+		it('should throw when nodeId is empty string', () => {
+			expect(() => new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY, '')).toThrow();
+		});
+	});
+
+	describe('fromFigmaDesignUrl', () => {
+		it('should return identity when Atlassian design ID does not contain node part', () => {
+			const result = FigmaDesignIdentity.fromAtlassianDesignId(
+				`${MOCK_FIGMA_FILE_KEY}/${MOCK_FIGMA_NODE_ID}`,
+			);
+
+			expect(result).toStrictEqual(
+				new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY, MOCK_FIGMA_NODE_ID),
+			);
+		});
+
+		it('should return identity when Atlassian design ID contains node part', () => {
+			const result = FigmaDesignIdentity.fromAtlassianDesignId(
+				`${MOCK_FIGMA_FILE_KEY}/${MOCK_FIGMA_NODE_ID}`,
+			);
+
+			expect(result).toStrictEqual(
+				new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY, MOCK_FIGMA_NODE_ID),
+			);
+		});
+
+		it('should throw when  Atlassian design ID has unexpected format', () => {
+			expect(() => FigmaDesignIdentity.fromFigmaDesignUrl(``)).toThrow();
```

Comment:
Should be calling `fromAtlassianDesignId`

Reply:
Resolved!